File: models/Tournament.ts
import mongoose, { Schema, model, models } from 'mongoose';

const TournamentSchema = new Schema({
  id: { type: String, required: true, unique: true },
  title: { type: String, required: true },
  game: { type: String, enum: ['BGMI', 'Free Fire'], required: true },
  prizePool: { type: String, required: true },
  date: { type: String, required: true },
  time: { type: String, required: true },
  slots: { type: String, required: true },
  image: { type: String, required: true },
  status: { type: String, enum: ['Open', 'Closed', 'Coming Soon'], default: 'Open' },
  allowedMatchTypes: { type: [String], default: ['Solo', 'Duo', 'Squad'] },
  createdAt: { type: Date, default: Date.now },
});

// Force schema update in development
if (mongoose.models.Tournament) {
  delete mongoose.models.Tournament;
}

export const Tournament = model('Tournament', TournamentSchema);

File: tournament_app/lib/models/tournament.dart
class Tournament {
  final String id;
  final String title;
  final String game;
  final String prizePool;
  final String date;
  final String time;
  final String slots;
  final String image;
  final String status;
  final List<String> allowedMatchTypes;

  Tournament({
    required this.id,
    required this.title,
    required this.game,
    required this.prizePool,
    required this.date,
    required this.time,
    required this.slots,
    required this.image,
    required this.status,
    required this.allowedMatchTypes,
  });

  factory Tournament.fromJson(Map<String, dynamic> json) {
    var allowedListRaw = json['allowedMatchTypes'];
    List<String> matchTypes = [];
    if (allowedListRaw is List) {
      matchTypes = allowedListRaw.map((e) => e.toString()).toList();
    } else {
      matchTypes = ['Solo', 'Duo', 'Squad'];
    }

    return Tournament(
      id: json['id'] ?? '',
      title: json['title'] ?? '',
      game: json['game'] ?? 'BGMI',
      prizePool: json['prizePool'] ?? '',
      date: json['date'] ?? '',
      time: json['time'] ?? '',
      slots: json['slots'] ?? '0/48',
      image: json['image'] ?? '',
      status: json['status'] ?? 'Open',
      allowedMatchTypes: matchTypes,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'game': game,
      'prizePool': prizePool,
      'date': date,
      'time': time,
      'slots': slots,
      'image': image,
      'status': status,
      'allowedMatchTypes': allowedMatchTypes,
    };
  }
}
